content/blog/2020-12-19-LLVM-IR-to-assembly.md

```diff
@@ -0,0 +1,76 @@
+
+# Hardest Part of the Project?
+X86-64 is the hardest part of the project.  Although it is widely adopted and the codebase for x86 in LLVM relatively well maintained, the x86-64 ISA is so complex and riddled with special rules and cases that makes any simple modification very painful to implement.  It is furthermore complicated by the complexity of the different concurrent calling conventions that x86-64 practices.  Going back to our recent class discussion about rebuild or fix, I think x86 is really something we need consider razing and rebuilding from scratch.  Not only is x86 ISA the problem, but the current LLVM codebase in x86 is full of legacy code and code to emulate “bugs” (or features) in GCC compiler.  Although this dates to the [history of compilers](https://sylvestre.ledru.info/blog/2014/09/11/rebuild-of-debian-using-clang-3-5), this makes x86-64 very unfriendly for any new developers to work on it.  Another notable mention is the amount of effort it takes to demystify the target machine files in LLVM.  
+
+# Results and Success?
+The project is a success in that I am able to report the performance compression with both microbenchmarks and on a FFT benchmark with varying number of architectural registers. 
+On the microbenchmarks, I have choosen to compare the number of architectural registers of matrix multiplication in scalar execution, SEE, AVX4 and AVX8 on x86 machine.  For each one of the implementations, they are being compiled by a unmodified version of GCC, unmodified version of clang, and a modified version of clang with only 75% of the original x86-64 registers.    
+The benchmark is primarily done by FFT and is implemented in SSE.   Thie code is compiled by three diffeent compilers: a unmodified version of GCC, unmodified version of clang, and a modified version of clang with only 75% of the original x86-64 registers.     
+The benchmark is done on a 4 core i9-9900K CPU with 16 GB RAM in a Ubuntu VM.  The results are reported cycle count with rdtsc and execution time is reported as wall-clock time.  The results are verified by comparing the results of different implementations.
```

Comment:
Maybe these are intended to be separate paragraphs too, requiring blank lines? You can preview the Markdown in using Zola if you're unsure how it will look.